utils/wasm-gen/src/lib.rs

```diff
@@ -236,25 +242,95 @@ pub fn gen_wasm_smith_module(u: &mut Unstructured, config: &SwarmConfig) -> Modu
     }
 }
 
+fn build_checked_call(
+    u: &mut Unstructured,
+    results: &[ValueType],
+    params_rules: &[Parameter],
+    func_no: u32,
+    memory_pages: WasmPageCount,
+    unchecked_memory: Ratio,
+) -> Vec<Instruction> {
+    let unchecked = unchecked_memory.get(u);
+
+    let mut code = Vec::with_capacity(params_rules.len() * 2 + 1 + results.len());
+    for parameter in params_rules {
+        match parameter {
+            Parameter::Value { value_type, rule } => {
+                let instr = match value_type {
+                    ValueType::I32 => Instruction::I32Const(rule.get_i32(u)),
+                    ValueType::I64 => Instruction::I64Const(rule.get_i64(u)),
+                    _ => panic!("Cannot handle f32/f64"),
+                };
+                code.push(instr);
+            }
+
+            Parameter::MemoryArray => {
+                if unchecked {
+                    code.push(Instruction::I32Const(u.arbitrary().unwrap()));
+                    code.push(Instruction::I32Const(u.arbitrary().unwrap()));
+                } else {
+                    let memory_size = memory_pages.size();
+                    let upper_limit = memory_size.saturating_sub(1);
+
+                    let pointer_beyond = u.int_in_range(0..=upper_limit).unwrap();
+                    let offset = u.int_in_range(0..=pointer_beyond).unwrap();
+
+                    code.push(Instruction::I32Const(offset as i32));
+                    code.push(Instruction::I32Const((pointer_beyond - offset) as i32));
+                }
+            }
+
+            Parameter::MemoryValue => {
+                if unchecked {
+                    code.push(Instruction::I32Const(u.arbitrary().unwrap()));
+                } else {
+                    let memory_size = memory_pages.size();
+                    // u/i128 are the biggest values with 8 bytes size.
```

Comment:
what about our repr(C) structs from gsys? e.g. LengthWithTwoHashes, that uses 32+32+4 = 68 bytes?

Reply:
good catch, thanks! Will use some hard-coded value, 100 for example.